tests/helpers/kube.go

```diff
@@ -1255,3 +1258,47 @@ func PortForwardPod(ctx context.Context, restConfig *rest.Config, namespace, pod
 
 	return stopChan, readyChan
 }
+
+// WaitForNodeConditionWithCheckName waits for the node to have a condition with the reason as checkName.
+func WaitForNodeConditionWithCheckName(ctx context.Context, t *testing.T, c klient.Client, nodeName, checkName, message string) {
```

Comment:
Should we wrap the function referenced below around require.Eventually? rather than re-writing the whole logic again?